### PR TITLE
shooter simulation

### PR DESCRIPTION
**Pull request by greenblitz4590**
😎👌

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 1]
```diff
@@ -27,7 +28,7 @@ private OI() {
         mainJoystick = new SmartJoystick(RobotConstants.Joystick.MAIN);
         secondJoystick = new SmartJoystick(RobotConstants.Joystick.SECOND);
         initButtons();
-        initializeDefaultCommands();
+//        initializeDefaultCommands();
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/OI.java:31]: commented code?

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 2]
```diff
@@ -46,7 +47,29 @@ public SmartJoystick getSecondJoystick() {
...
+            @Override
+            public void end(boolean interrupted) {
+                pivot.setPower(0);
+            }
+        });
+
+        mainJoystick.A.whileTrue(new PivotCommand(){
+            @Override
+            public void execute() {
+                pivot.setPower(-0.2);
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/OI.java:65]: delete

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 3]
```diff
@@ -46,7 +47,29 @@ public SmartJoystick getSecondJoystick() {
...
+            }
+
+            @Override
+            public void end(boolean interrupted) {
+                pivot.setPower(0);
+            }
+        });
     }
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/OI.java:73]: also this

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Mechanism/ShooterMechanism.java [hunk 4]
```diff
@@ -42,11 +42,9 @@ private ShooterMechanism(){
 
 
     public void periodic() {
+        pivot.setAngle(-Pivot.getInstance().getAngle().getDegrees());
 
-        double pivotAngle = Pivot.getInstance().getAngleInRadians();
-        pivot.setAngle(Units.radiansToDegrees(pivotAngle));
-
-        Logger.recordOutput("Pivot/SimPose3D", Pivot.getInstance().getAngleInRadians());
+        Logger.recordOutput("Pivot/SimPose3D", Pivot.getInstance().getAngle());
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Mechanism/ShooterMechanism.java:47]: delete, need only at th PivotSubsysytem

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Mechanism/ShooterMechanism.java [hunk 5]
```diff
@@ -42,11 +42,9 @@ private ShooterMechanism(){
 
 
     public void periodic() {
+        pivot.setAngle(-Pivot.getInstance().getAngle().getDegrees());
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Mechanism/ShooterMechanism.java:45]: why minus

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Mechanism/ShooterMechanism.java (hunk 5 above)
Comment (YoniKiriaty) [reply src/main/java/edu/greenblitz/robotName/subsystems/shooter/Mechanism/ShooterMechanism.java:45]: because the mechanism 2d has opposite axes

**Review by aaa-png — APPROVED** [review]

**Review by NoamRosenberg08 — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/Pivot.java [hunk 6]
```diff
@@ -90,13 +87,18 @@ public double getVelocity() {
...
-        return Math.abs(angle - getAngleInRadians()) <= TOLERANCE;
+
+    public boolean isAtAngle(Rotation2d angle) {
+        return Math.abs(angle.getRadians() - getAngle().getRadians()) <= TOLERANCE;
     }
-    
 
+    public Pose3d getPivotPose() {
+        return new Pose3d(
+                new Translation3d(0,0.1,0.15),
```
Comment (tnrz135) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/Pivot.java:100]: Magic

**Review by tnrz135 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Mechanism/ShooterMechanism.java
Comment (tnrz135) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Mechanism/ShooterMechanism.java:1]: This file is empty

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/Pivot.java [hunk 7]
```diff
@@ -93,13 +90,18 @@ public double getVelocity() {
...
-	}
+    public Rotation2d getAngle() {
+        return Rotation2d.fromRadians(pivotInputs.position);
+    }
 
+    public boolean isAtAngle(Rotation2d angle) {
+        return Math.abs(angle.getRadians() - getAngle().getRadians()) <= TOLERANCE;
+    }
 
+    public Pose3d getPivotPose() {
```
Comment (tnrz135) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/Pivot.java:101]: image
I heard this is a great show, but why do we care about it's pivot?

**Review by tnrz135 — CHANGES_REQUESTED** [review]

**Review by tnrz135 — APPROVED** [review]

**Comment by bossyshrimp68** [comment]
יוני המשיח

**Pull request merged**